api/app/db/crud/weather_models.py

```diff
@@ -171,77 +160,63 @@ def get_latest_station_model_prediction_per_day(session: Session,
-        WeatherStationModelPrediction.tmp_tgl_2,
-        WeatherStationModelPrediction.bias_adjusted_temperature,
-        WeatherStationModelPrediction.bias_adjusted_rh,
-        WeatherStationModelPrediction.apcp_sfc_0,
-        WeatherStationModelPrediction.wdir_tgl_10,
-        WeatherStationModelPrediction.wind_tgl_10,
-        WeatherStationModelPrediction.update_date)\
-        .join(PredictionModelRunTimestamp, WeatherStationModelPrediction.prediction_model_run_timestamp_id == PredictionModelRunTimestamp.id)\
-        .join(PredictionModel, PredictionModelRunTimestamp.prediction_model_id == PredictionModel.id)\
-        .join(subquery, and_(
-            WeatherStationModelPrediction.prediction_timestamp == subquery.c.latest_prediction,
-            WeatherStationModelPrediction.station_code == subquery.c.station_code))\
-        .filter(PredictionModel.abbreviation == model)\
-        .order_by(WeatherStationModelPrediction.update_date.desc())
     return result
 
 
-def get_latest_station_prediction_mat_view(session: Session,
-                                           station_codes: List[int],
-                                           day_start: datetime.datetime,
-                                           day_end: datetime.datetime):
-    logger.info("Getting data from materialized view.")
-    result = session.query(MoreCast2MaterializedView.prediction_timestamp,
-                           MoreCast2MaterializedView.abbreviation,
-                           MoreCast2MaterializedView.station_code,
-                           MoreCast2MaterializedView.rh_tgl_2,
-                           MoreCast2MaterializedView.tmp_tgl_2,
-                           MoreCast2MaterializedView.bias_adjusted_temperature,
-                           MoreCast2MaterializedView.bias_adjusted_rh,
-                           MoreCast2MaterializedView.bias_adjusted_wind_speed,
-                           MoreCast2MaterializedView.bias_adjusted_wdir,
-                           MoreCast2MaterializedView.precip_24h,
-                           MoreCast2MaterializedView.bias_adjusted_precip_24h,
-                           MoreCast2MaterializedView.wdir_tgl_10,
-                           MoreCast2MaterializedView.wind_tgl_10,
-                           MoreCast2MaterializedView.update_date).\
-        filter(MoreCast2MaterializedView.station_code.in_(station_codes),
-               MoreCast2MaterializedView.prediction_timestamp >= day_start,
-               MoreCast2MaterializedView.prediction_timestamp <= day_end)
-    return result
-
-
-def get_latest_station_prediction_per_day(session: Session,
-                                          station_codes: List[int],
-                                          day_start: datetime.datetime,
-                                          day_end: datetime.datetime):
+def get_latest_station_prediction_per_day(session: Session, station_codes: List[int], day_start: datetime.datetime, day_end: datetime.datetime):
```

Comment:
Would you mind removing this function? Is doesn't look like it is used anywhere.

Reply:
Done in: https://github.com/bcgov/wps/pull/4118/commits/3d8ad27a8f7a39d5c19d15f0ed316fbaaa279fb9

Which made the dedup'ing of the subquery moot, so I folded that code back in.